floodviz/static/js/map.js

```diff
@@ -1,193 +1,357 @@
-					if (property_for_id && d.properties[property_for_id]) {
-						return 'map' + d.properties[property_for_id];
+		FV.mapmodule = function (options) {
+
+			var self = {};
+
+			// Stores SVG coordinates of gages and the size and location of the selection box
+			var state = {};
+
+			var project = function (lambda, phi) {
+				return options.proj.forward([lambda, phi].map(radiansToDegrees));
+			};
+			project.invert = function (x, y) {
+				return options.proj.inverse([x, y]).map(degreesToRadians);
+			};
+			//Define map projection
+			var projection = d3.geoProjection(project);
+			// Give projection initial rotation and scale
+			projection.scale(1).translate([0, 0]);
+			//Define path generator
+			var path = d3.geoPath().projection(projection);
+			//Create SVG element
+			var svg = null;
+			// Tooltip
+			var maptip = d3.select("body")
+				.append("div")
+				.attr("id", "maptip");
+
+			/**
+			 * Add circles to the map.
+			 * @param data The geojson to be added to the svg
+			 * @param classname The class to be given to each element for use in CSS
+			 * @param radius The radius of each circle. This cannot be set from CSS
+			 * @param property_for_id The name of a field in the 'properties' of each feature, to be used for ID
+			 *                            If null, or not provided, no id will be given.
+			 *
+			 * @return group The group that has been appended to the SVG
+			 */
+			var add_circles = function (data, classname, radius, property_for_id) {
+				var group = svg.append("g");
+				group.selectAll("circle")
+					.data(data.features)
+					.enter()
+					.append("circle")
+					.attr("r", radius)
+					.attr("transform", function (d) {
+						return "translate(" + projection(d.geometry.coordinates) + ")";
+					})
+					.attr("id", function (d) {
+						if (property_for_id && d.properties[property_for_id]) {
+							return 'map' + d.properties[property_for_id];
+						}
+						else {
+							return '';
+						}
+					})
+					.attr("class", classname)
+					.each(function (d) {
+						if (property_for_id && d.properties[property_for_id]) {
+							if (FV.hydrograph_display_ids.indexOf(d.properties.id) !== -1) {
+								self.site_add_accent(d.properties.id);
+							}
+						}
+					});
+				return (group);
+			};
+			/**
+			 * Add paths to the map
+			 * @param data The geojson to be added to the svg
+			 * @param classname The class to be given to each element for use in CSS
+			 *
+			 * @return group The group that has been appended to the SVG
+			 */
+			var add_paths = function (data, classname) {
+				var group = svg.append("g");
+				group.selectAll("path")
+					.data(data.features)
+					.enter()
+					.append("path")
+					.attr("d", path)
+					.attr("class", classname);
+				return group;
+			};
+
+			/**
+			 * Add or remove the line corresponding to a gage from the hydrograph
+			 * @param sitekey THe ID of the gage in question
+			 */
+			var toggle_hydrograph_display = function (sitekey) {
```

Comment:
It seems like we are targeting the hydrograph specifically again. We may need to revisit the linked_interactions so that the linked_nteractions are abstracted better. This is okay for now.